DataFormats/TrackingRecHitSoA/test/alpaka/Hits_test.cc

```diff
@@ -34,33 +37,68 @@ int main() {
     {
       uint32_t nHits = 2000;
       int32_t offset = 100;
-      auto moduleStartH =
-          cms::alpakatools::make_host_buffer<uint32_t[]>(queue, pixelTopology::Phase1::numberOfModules + 1);
-      for (size_t i = 0; i < pixelTopology::Phase1::numberOfModules + 1; ++i) {
+      uint32_t nModules = 200;
+
+      SiPixelClustersSoACollection clusters(nModules, queue);
+      clusters.setNClusters(nHits, offset);
+
+      auto moduleStartH = cms::alpakatools::make_host_buffer<uint32_t[]>(queue, nModules + 1);
+
+      for (size_t i = 0; i < nModules + 1; ++i) {
         moduleStartH[i] = i * 2;
       }
-      auto moduleStartD =
-          cms::alpakatools::make_device_buffer<uint32_t[]>(queue, pixelTopology::Phase1::numberOfModules + 1);
+
+      auto hitsX = cms::alpakatools::make_host_buffer<float[]>(queue, nHits);
+      for (size_t i = 0; i < nHits; ++i) {
+        hitsX[i] = float(i) * 2;
+      }
+
+      auto moduleStartD = cms::alpakatools::make_device_view<uint32_t>(queue, clusters.view().clusModuleStart(), nHits);
       alpaka::memcpy(queue, moduleStartD, moduleStartH);
-      TrackingRecHitsSoACollection<pixelTopology::Phase1> tkhit(queue, nHits, offset, moduleStartD.data());
 
-      testTrackingRecHitSoA::runKernels<pixelTopology::Phase1>(tkhit.view(), queue);
+      TrackingRecHitsSoACollection tkhit(queue, clusters);
+
+      // exercise the copy of a full column (on device)
+      auto hitXD = cms::alpakatools::make_device_view<float>(queue, tkhit.view().xLocal(), nHits);
+      alpaka::memcpy(queue, hitXD, hitsX);
+
+      // exercise the memset of a colum (on device)
+      auto hitYD = cms::alpakatools::make_device_view<float>(queue, tkhit.view().yGlobal(), nHits);
+      constexpr float constYG = -14.0458;
+      std::vector<float> constXV(nHits, constYG);
+      auto constYGV_v = cms::alpakatools::make_host_view<float>(constXV.data(), nHits);
+      alpaka::memcpy(queue, hitYD, constYGV_v);
```

Comment:
This _should_ also work:
```suggestion
      alpaka::memcpy(queue, hitYD, constXV);
```

Reply:
Apparently it doesn't like 

```c++
 std::vector<float> constXV(nHits, constYG);
 alpaka::memcpy(queue, hitYD, constXV);
 ```
 
 and complains with
 
 ```bash
 In file included from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/mem/buf/Traits.hpp:10,
                 from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/dev/DevCpu.hpp:11,
                 from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/acc/AccCpuOmp2Blocks.hpp:37,
                 from /cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/alpaka.hpp:13,
                 from src/DataFormats/TrackingRecHitSoA/test/alpaka/Hits_test.cc:4:
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/mem/view/Traits.hpp: In instantiation of 'auto alpaka::createTaskMemcpy(TViewDstFwd&&, const TViewSrc&, const TExtent&) signed int, 1>, long unsigned int>; TViewSrc = std::vector<float>; TViewDstFwd = ViewPlainPtr<DevCpu, float, std::integral_constant<long unsigned int, 1>, unsigned int>&]':                                                                             
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/mem/view/Traits.hpp:309:40:   required from 'void alpaka::memcpy(TQueue&, TViewDstFwd&&, const TViewSrc&) [with TViewSrcDevCpu, float, std::integral_constant<long unsigned int, 1>, unsigned int>&; TQueue = QueueGenericThreadsBlocking<DevCpu>]'
src/DataFormats/TrackingRecHitSoA/test/alpaka/Hits_test.cc:70:21:   required from here
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/mem/view/Traits.hpp:274:58: error: static assertion failed: The destination view and the extent are required to have com
  274 |             meta::IsIntegralSuperset<DstIdx, ExtentIdx>::value,
      |                                                          ^~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02882/el8_amd64_gcc12/external/alpaka/1.2.0-0ef9ed9c17707f526f6e6d8263e2898d/include/alpaka/mem/view/Traits.hpp:274:58: note: 'std::integral_constant<bool, false>::value' evaluates to false
```
 
 :-/

Reply:
Ah... alpaka does not like it because the `vector` uses `unsigned long` as size type, while the destination view uses `unsigned int`, and so the `vector` could potentially have more elements that can be stored in the destination view size.

I've opened https://github.com/alpaka-group/alpaka/issues/2491 to see if we can relax the compile time check.

Reply:
Thanks